iq/iq.go

```diff
@@ -502,17 +520,42 @@ func (i *Server) pollIQServer(statusURL string, finished chan resultError) error
 				Message: "Could not unmarshal response from IQ server",
 			}
 		}
+
+		i.logLady.WithFields(logrus.Fields{
+			"response": response,
+		}).Trace("Nexus IQ polling response")
+
 		statusURLResp = response
 		if response.IsError {
 			finished <- resultError{finished: true, err: nil}
 		}
+
+		i.populateAbsoluteURL()
 		finished <- resultError{finished: true, err: nil}
 	}
 	i.tries++
 	fmt.Print(".")
 	return err
 }
 
+func (i *Server) populateAbsoluteURL() {
```

Comment:
You might actually flip this from a method off of `(i *Server)` to a method off of `(s StatusURLResult)`, then you could just access it as a method. Buyers choice, but it's working with that data more so than it is `Server`.

Reply:
> You might actually flip this from a method off of `(i *Server)` to a method off of `(s StatusURLResult)`, then you could just access it as a method. Buyers choice, but it's working with that data more so than it is `Server`.

I was on the fence about this, due to the need to access the `Server` field to get the base url. I moved it as suggested and I think it is an improvement. Decided to pass in the base url to make it easier to test and possibly easier to reuse elsewhere. ca41533